solution for longest ZigZag path in binary tree

Shruti-425 opened the pull request:
Implement a solution to find the longest ZigZag path in a binary tree. The approach uses a recursive function to calculate the lengths of ZigZag paths from each node.

Shruti-425 committed:
solution for longest ZigZag path in binary tree

Implement a solution to find the longest ZigZag path in a binary tree. The approach uses a recursive function to calculate the lengths of ZigZag paths from each node.

abhisek247767 reviewed (changes requested):
commit the code on c++ folder @Shruti-425

Shruti-425 committed:
Add solution for longest ZigZag path in binary tree

Implement a solution to find the longest ZigZag path in a binary tree, utilizing a recursive approach to calculate left and right lengths.

Shruti-425 committed:
Merge branch 'abhisek247767:main' into main
